### PR TITLE
deploymentExists different from component running

### LINKED ISSUE #6022: odo is not executing `run` command on java-quarkus devfile
```
▶ odo init --name test --devfile java-quarkus --starter community

  __
 /  \__     Initializing new component
 \__/  \
 /  \__/    odo version: v3.0.0-beta3
 \__/

Interactive mode enabled, please answer the following questions:
 ✓  Downloading devfile "java-quarkus" [2s]
 ✓  Downloading starter project "community" [872ms]

Your new component 'test' is ready in the current directory.
To start editing your component, use 'odo dev' and open this folder in your favorite IDE.
Changes will be directly reflected on the cluster.


▶ odo dev
  __
 /  \__     Developing using the test Devfile
 \__/  \    Namespace: tkral-dev
 /  \__/    odo version: v3.0.0-beta3
 \__/

↪ Deploying to the cluster in developer mode
 •  Waiting for Kubernetes resources  ...
 ✓  Added storage m2 to
 ⚠  Pod is Pending
 ✓  Pod is Running
 ✓  Syncing files into the container [2s]
 ✓  Executing init-compile command "mvn -Dmaven.repo.local=/home/user/.m2/repository compile" on container "tools" [36s]
 -  Forwarding from 127.0.0.1:40001 -> 8080


Watching for changes in the current directory /Users/tkral/Code/odo-examples/quarkus-test
Press Ctrl+c to exit `odo dev` and delete resources from the cluster

E0811 16:35:52.106701   26001 portforward.go:406] an error occurred forwarding 40001 -> 8080: error forwarding port 8080 to pod 3ab7daf467c6f7a573bc63d70080e4a0e5b1b80fbd747eb99a22324752153555, uid : port forward into network namespace "/var/run/netns/ba0ee200-62e8-45ff-afbc-224304a0b660": failed to connect to localhost:8080 inside namespace 3ab7daf467c6f7a573bc63d70080e4a0e5b1b80fbd747eb99a22324752153555: dial tcp [::1]:8080: connect: connection refused
E0811 16:35:52.109314   26001 portforward.go:234] lost connection to pod
 -  Forwarding from 127.0.0.1:40001 -> 8080



▶ curl localhost:40001
curl: (52) Empty reply from server

```

There is no information in `odo dev` output that the run command would be executed, and the output of the `odo logs` suggests the same.

```
▶ odo logs 
...
...
...
tools: [INFO] Changes detected - recompiling the module!
tools: [INFO] Compiling 2 source files to /projects/target/classes
tools: [INFO] ------------------------------------------------------------------------
tools: [INFO] BUILD SUCCESS
tools: [INFO] ------------------------------------------------------------------------
tools: [INFO] Total time:  33.289 s
tools: [INFO] Finished at: 2022-08-11T14:35:18Z
tools: [INFO] ------------------------------------------------------------------------
```

```
▶ odo version
odo v3.0.0-beta3 (53a7c3cd1)
```

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind bug

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6022

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: tests/integration/cmd_dev_test.go [hunk 1]
```diff
@@ -2270,4 +2270,39 @@ CMD ["npm", "start"]
 			})
 		})
 	})
+
+	When("a hotRelad capable project is used with odo dev", func() {
```
Comment (valaparthvi) [inline tests/integration/cmd_dev_test.go:2274]:
```suggestion
	When("a hotReload capable project is used with odo dev", func() {
```

**Review by valaparthvi — CHANGES_REQUESTED** [review]
Can't we use the community starter project for testing instead of adding the whole project? Looking at the test case, it seems unnecessary to me.

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 2]
```diff
@@ -263,10 +263,10 @@ func (a Adapter) Push(parameters adapters.PushParameters, componentStatus *watch
...
 	var isComposite bool
 	cmdHandler := adapterHandler{
-		Adapter:         a,
-		parameters:      parameters,
-		componentExists: componentExists,
-		podName:         pod.GetName(),
+		Adapter:    a,
+		parameters: parameters,
+		//		componentExists:,
+		podName: pod.GetName(),
```
[the remark this answers is not in the document]
Comment (feloy) [reply pkg/devfile/adapters/kubernetes/component/adapter.go]: Thanks, I forgot this one

**Pull request merged**